scanner/mappers/mappers.go

```diff
@@ -788,3 +788,64 @@ func findName(vuln *claircore.Vulnerability, p *regexp.Regexp) (string, bool) {
 	}
 	return "", false
 }
+
+// filterVulns filters repeat vulnerabilities out of vulnIDs and returns the result.
+// This function does not guarantee ordering is preserved.
+func filterVulns(vulnIDs []string, vulns map[string]*claircore.Vulnerability) []string {
+	// Group each vulnerability by name.
+	// This maps each name to a slice of vulnerabilities to protect against the possibility
+	// Claircore finds multiple vulnerabilities with the same name for this package from different vulnerability streams.
+	// In that situation, it is not clear which one may be the single, correct option to choose, so just allow for both.
+	vulnsByName := make(map[string][]*claircore.Vulnerability)
+OUTER:
+	for _, vulnID := range vulnIDs {
+		vuln := vulns[vulnID]
+		if vuln == nil {
+			continue
```

Comment:
That would mean a bug, right? A vulnerability is associated with a package that does not have an object in the vulnerability mapping. Should we just ignore it?

Reply:
I think this is ok. It probably won't happen, but I like to be extra safe and check for `nil`. Not sure if we currently have a better place to do any kind of sanity checks beforehand, and adding that to this PR may be out of scope